### PR TITLE
[Helm/Release] Add examples to API reference docs and sdks

### LINKED ISSUE #1695: [Helm/Release] Add examples into description/schema in each language for docs

### PR DESCRIPTION
**Pull request by viveklak**
### Proposed changes

This change adds a number of reference examples for using Helm Release resource to API docs and SDK docstrings.

### Related issues (optional)
Fixes #1695

### CODE REVIEW

File: provider/pkg/gen/examples/overlays/helmRelease.md [hunk 1]
```diff
@@ -1,2 +1,746 @@
...
+			},
+			ValueYamlFiles: pulumi.NewFileAsset("./metrics.yml"),
+			Value: pulumi.Map{
+				"cluster": pulumi.Map{
+                    "enabled": pulumi.Bool(true),
+                },
+                "rbac": pulumi.Map{
+                    "create": pulumi.Bool(true),
+                },
+			},
```
Comment (lblackstone) [inline provider/pkg/gen/examples/overlays/helmRelease.md:580]: Looks like the indentation is wrong here

**Review by lblackstone — APPROVED** [review]
Looks great aside from the one formatting comment 🎉

File: provider/pkg/gen/examples/overlays/helmRelease.md (hunk 1 above)
Comment (viveklak) [reply provider/pkg/gen/examples/overlays/helmRelease.md]: Thanks for pointing out! Fixed now.

File: provider/pkg/gen/examples/overlays/helmRelease.md [hunk 2]
```diff
@@ -1,2 +1,746 @@
...
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		_, err := helm.NewRelease(ctx, "redis", helm.ReleaseArgs{
+			Chart:   pulumi.String("redis"),
+			RepositoryOpts: helm.RepositoryOptsArgs{
+				Repo: pulumi.String("https://charts.helm.sh/stable"),
+			},
+			ValueYamlFiles: pulumi.NewFileAsset("./metrics.yml"),
+			Value: pulumi.Map{
+				"cluster": pulumi.Map{
+                    "enabled": pulumi.Bool(true),
```
Comment (lblackstone) [inline provider/pkg/gen/examples/overlays/helmRelease.md:575]: Another indent error here

Comment (viveklak) [reply provider/pkg/gen/examples/overlays/helmRelease.md]: Grr. Goland mixing spaces/tabs... Fixed now I believe.

**Pull request merged**